libs/ledger-live-common/src/families/filecoin/bridge/utils/addresses.ts

```diff
@@ -32,3 +57,50 @@ export const validateAddress = (input: string): ValidateAddressResult => {
 
   return { isValid: false };
 };
+
+export const isRecipientValidForTokenTransfer = (addr: string): boolean => {
+  if (addr.length < 2) {
+    return false;
+  }
+
+  const valid = validateAddress(addr);
+  if (!valid.isValid) {
+    return false;
+  }
+
+  const accountType = addr.substring(0, 2);
+  if (["f0", "0x", "f4"].includes(accountType)) {
+    return true;
```

Comment:
if I understand, an address begins with "f0" or "f4", we should convert them into eth address and check whether it is a valid one. right?
we can't consider all the addresses begin with "f0" or "f4" are valid. right?

Reply:
you understand correct, for token transfer we need the address to be eth style (0x) not `f[0,1,4]`. In filecoin ecosystem there's ways to convert all native addresses to eth style but f1 is exception which requires network call (insecure) to get equivalent f0 and then f0 -> 0x (secure). 

For addresses beginning with f1 we do not want to support them for token transfer because network call can be insecure in practice and lead to possible known attacks. For when invalid we therefore have this [error](https://github.com/LedgerHQ/ledger-live/blob/a4344ecb8d7e980c48c6b450b7021119681ffa8e/apps/ledger-live-desktop/static/i18n/en/app.json#L5498).